### PR TITLE
restruct Identity with enum(fix #1051)

### LINKED ISSUE #1051: Restructure or add sanity check of Identity
### Context
F/E colleagues have discovered a problem that any combination of `web_type` and `handle` is allowed, e.g. you can use `Twitter` and `Address32`, which is actually wrong.

It's better to prevent this from parachain side, we need to either:
- do sanity checks and throw an error
- force it by restructuring. e.g. something like `Identity::Twitter(PlainString)`

Better ideas?

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by DogLi**
restruct `Identiy` with enum to avoid the free combination of network type and handle type

### CODE REVIEW

File: tee-worker/litentry/primitives/src/lib.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@
 // along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
 
 #![cfg_attr(not(feature = "std"), no_std)]
+extern crate core;
```
Comment (Kailai-Wang) [inline tee-worker/litentry/primitives/src/lib.rs]: I believe this is redundant?

File: mock-tee-primitives/src/identity.rs [hunk 2]
```diff
@@ -67,97 +81,128 @@ pub enum Web2Network {
...
 
 #[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-pub struct Identity {
-	pub web_type: IdentityWebType,
-	pub handle: IdentityHandle,
+pub enum Identity {
+	Substrate(SubstrateIdentity),
+	Evm(EvmIdentity),
+	Web2(Web2Identity),
```
Comment (Kailai-Wang) [inline mock-tee-primitives/src/identity.rs]: I'm basically fine with it, how does this new structure sound to you?  @zhizming-zhong @Verin1005 @jonalvarezz 

Btw I'm just thinking if we should simplify it even more to flatten out the `SubstrateIdentity`/`EvmIdentity`/`Web2Identity`, so something like this:
```
	Substrate(SubstrateNetwork, Address32),
	Evm(EvmNetwork, Address20),
	Web2(Web2Network, IdentityString),
```

**Review by Kailai-Wang — APPROVED** [review]
Basically looks good to me, thanks. 

But about the new `Identity` enum structure, I'd like to hear opinions from F/E colleagues.

edit: CI should be fixed though

File: tee-worker/litentry/primitives/src/lib.rs (hunk 1 above)
Comment (DogLi) [reply tee-worker/litentry/primitives/src/lib.rs]: it's auto-added by the IDE, I'll fix this.

File: mock-tee-primitives/src/identity.rs (hunk 2 above)
Comment (ziming-zung) [reply mock-tee-primitives/src/identity.rs]: I prefer the flatter one

Comment (jingleizhang) [reply mock-tee-primitives/src/identity.rs]: agree, the flatter version is more simpler

Comment (jonalvarezz) [reply mock-tee-primitives/src/identity.rs]: Thank you, Kai.

What matters the most to clients is how easy it is to manipulate and traverse the objects.

Do I understand correctly that the following will be the corresponding Polkadot.js API types?

For `Substrate(SubstrateNetwork, Address32)`
```js
Identity: {
  Substrate: [SubstrateNetworkObject, '0x1234....']
}
```

For `Substrate(SubstrateNetworkIdentity)`
```js
Identity: {
  Substrate: {
    network: '...',
    address: '0x1234....'
  }
}
```

If so, we would prefer `Identity.Substrate.network` over `Identity.Substrate[1]`. Object properties over array indexes, in general.

Comment (Kailai-Wang) [reply mock-tee-primitives/src/identity.rs]: I actually think they should have the same polkadot-js types (i.e. without array indexes), but please correct me if I'm wrong @Verin1005 @zhizming-zhong

Comment (ziming-zung) [reply mock-tee-primitives/src/identity.rs]: Yes, if the order of their fields is the same, they have the same value after encoding

**Review by jingleizhang — APPROVED** [review]
LGTM

File: mock-tee-primitives/src/identity.rs (hunk 2 above)
Comment (jonalvarezz) [reply mock-tee-primitives/src/identity.rs]: If that's the case, then there is not much difference from our side :)

The plain version looks easier to follow, though 👍

**Pull request merged**